pyomo/contrib/solver/results.py

```diff
@@ -33,107 +33,94 @@ class TerminationCondition(enum.Enum):
-        satisfied.
-    maxTimeLimit: 1
-        The solver exited due to reaching a specified time limit.
-    iterationLimit: 2
-        The solver exited due to reaching a specified iteration limit.
-    objectiveLimit: 3
-        The solver exited due to reaching an objective limit. For example,
-        in Gurobi, the exit message "Optimal objective for model was proven to
-        be worse than the value specified in the Cutoff parameter" would map
-        to objectiveLimit.
-    minStepLength: 4
-        The solver exited due to a minimum step length.
-        Minimum step length reached may mean that the problem is infeasible or
-        that the problem is feasible but the solver could not converge.
-    unbounded: 5
-        The solver exited because the problem has been found to be unbounded.
-    provenInfeasible: 6
-        The solver exited because the problem has been proven infeasible.
-    locallyInfeasible: 7
-        The solver exited because no feasible solution was found to the
-        submitted problem, but it could not be proven that no such solution exists.
-    infeasibleOrUnbounded: 8
-        Some solvers do not specify between infeasibility or unboundedness and
-        instead return that one or the other has occurred. For example, in
-        Gurobi, this may occur because there are some steps in presolve that
-        prevent Gurobi from distinguishing between infeasibility and unboundedness.
-    error: 9
-        The solver exited with some error. The error message will also be
-        captured and returned.
-    interrupted: 10
-        The solver was interrupted while running.
-    licensingProblems: 11
-        The solver experienced issues with licensing. This could be that no
-        license was found, the license is of the wrong type for the problem (e.g.,
-        problem is too big for type of license), or there was an issue contacting
-        a licensing server.
-    emptyModel: 12
-        The model being solved did not have any variables
-    unknown: 42
-        All other unrecognized exit statuses fall in this category.
     """
 
     convergenceCriteriaSatisfied = 0
+    "The solver exited because convergence criteria of the problem were satisfied."
 
     maxTimeLimit = 1
+    """The solver exited due to reaching a specified time limit."""
 
     iterationLimit = 2
+    """The solver exited due to reaching a specified iteration limit."""
 
     objectiveLimit = 3
+    """The solver exited due to reaching an objective limit. For example, in
+    Gurobi, the exit message "Optimal objective for model was proven to
+    be worse than the value specified in the Cutoff parameter" would map
+    to objectiveLimit.
+    """
 
     minStepLength = 4
+    """The solver exited due to a minimum step length.  Minimum step length
+    reached may mean that the problem is infeasible or that the problem
+    is feasible but the solver could not converge.
+    """
```

Comment:
Are there negative implications to moving these descriptions out of the class docstring? For example, if a user calls `help` on one of these enum classes will they see all the available options?

Reply:
I'd prefer that the help messages stay for this one purely because it is new / we want visibility into what the new solver interface is doing.